### PR TITLE
release/1.5.0

### PR DESCRIPTION
**Pull request by datawrestler**
### v1.5.0
- Performance improvement related to data processing and memory usage for `property.search`.
- `property.events` now supports concurrent requests. 
- Introduce client parameter `num_workers` to control the number of concurrent requests. Default is 10.

### CODE REVIEW

File: parcllabs/services/parcllabs_service.py [hunk 1]
```diff
@@ -199,17 +200,23 @@ def error_handling(self, response: requests.Response) -> None:
             error_details = response.json()
             error_message = error_details.get("detail", "No detail provided by API")
             error = error_message
+            
             if response.status_code == 403:
```
Comment (zhibindai26) [inline parcllabs/services/parcllabs_service.py:204]: could we create a function or method for handling the error codes and manage the different errors in there?

File: parcllabs/services/property_search.py [hunk 2]
```diff
@@ -131,159 +39,135 @@ def retrieve(
...
 
-        for parcl_id in parcl_ids:
-            params['parcl_id'] = parcl_id
-            headers = self._get_headers()
-            data = fetch_data(self.full_url, headers=headers, params=params)
-            tmp = pd.DataFrame()
+                for df_batch in process_data(
+                    data, batch_size=10000, num_workers=self.client.num_workers
+                ):
+                    tmp = pd.concat([tmp, df_batch], ignore_index=True)
```
Comment (zhibindai26) [inline parcllabs/services/property_search.py:133]: can we use a more descriptive var here?

**Comment by zhibindai26** [comment]
@datawrestler left a couple minor comments. Since the version is bumped, can we also change the PR name/description so its clear? Great work!

File: parcllabs/services/parcllabs_service.py (hunk 1 above)
Comment (datawrestler) [reply parcllabs/services/parcllabs_service.py:204]: kicking due to convo - method name is `error_handling` and currently accomplishes this.

**Review by zhibindai26 — APPROVED** [review]
big update!

**Pull request merged**